### PR TITLE
improve session related fuzzing

### PR DESCRIPTION
**Pull request by xiaotia3**
Revisited and improved following session related fuzzing - fuzz code fix, new seeds and coverage improvement:
- KEY_EXCHANGE/KEY_EXCHANGE_RSP
- FINISH/FINISH_RSP
- PSK_EXCHANGE/PSK_EXCHANGE_RSP
- PSK_FINISH/PSK_FINISH_RSP
- HEARTBEAT/HEARTBEAT_ACK
- KEY_UPDATE/KEY_UPDATE_ACK
- END_SESSION/END_SESSION_ACK

### CODE REVIEW

File: fuzz-target/fuzzlib/src/responderlib.rs [hunk 1]
```diff
@@ -38,6 +38,7 @@ pub fn rsp_create_info() -> (common::SpdmConfigInfo, common::SpdmProvisionInfo)
         aead_algo: SpdmAeadAlgo::AES_256_GCM,
         req_asym_algo: SpdmReqAsymAlgo::TPM_ALG_RSAPSS_2048,
         key_schedule_algo: SpdmKeyScheduleAlgo::SPDM_KEY_SCHEDULE,
+        secure_spdm_version: [0x11, 0x12],
```
Comment (xiaoyuxlu) [inline fuzz-target/fuzzlib/src/responderlib.rs:41]: We can drop SPDM1.1 for now.  Ref: https://github.com/jyao1/rust-spdm/issues/484

**Review by xiaoyuxlu — CHANGES_REQUESTED** [review]
I like fake_xxx functions for crypto functions. It will improve the efficiency of fuzz.
Can we register crypto functions out of xxxxx_fuzz(fuzz_data:&[u8])?  One registration is enough I think.

File: fuzz-target/fuzzlib/src/responderlib.rs (hunk 1 above)
Comment (xiaoyuxlu) [reply fuzz-target/fuzzlib/src/responderlib.rs:41]: I made a mistake, I'm sorry.
please ignore this comment

Comment (xiaoyuxlu) [reply fuzz-target/fuzzlib/src/responderlib.rs:41]: This field should select the versions of SPDM0277.

**Comment by xiaotia3** [comment]
> I like fake_xxx functions for crypto functions. It will improve the efficiency of fuzz. Can we register crypto functions out of xxxxx_fuzz(fuzz_data:&[u8])? One registration is enough I think.

Sometimes I need to fail hmac verification, it is more flexible to put it inside

File: fuzz-target/fuzzlib/src/responderlib.rs (hunk 1 above)
Comment (xiaotia3) [reply fuzz-target/fuzzlib/src/responderlib.rs]: Updated

**Review by xiaoyuxlu — APPROVED** [review]

**Comment by xiaoyuxlu** [comment]
> is this ready for merge?
Yes, it is.

**Pull request merged**